zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java

```diff
@@ -263,6 +263,7 @@ public final Builder dateSeparator(char dateSeparator) {
   public abstract int namesLookback();
 
   @Override
+  @Memoized // Memoized to reduce garbage as this is effectively a singleton.
```

Comment:
why. is this called per request? if so this also adds a lock internally. can we not add this at the moment until analyzed as this assumption isnt carried to the several other impls. the only other consumer of this is the http api and that isnt hit often enough for specializations.

Reply:
It's called per request. I noticed Cassandra memoizes these too, so figured doing similar would be ok. Was hoping to memoize at this point, since the constructors access multiple memoized fields themselves - but it's not a big deal so reverted.

Reply:
cassandra is different reason

```
  /** {@inheritDoc} Memoized in order to avoid re-preparing statements */
```

Right now the main attention is ideally drawn to important things like data or setup dependencies. I'm trying to help keep focus until things are working. I think this is really not going to affect performance in a holistic way to memoize because it is focusing on object allocation. Let's keep focus where it is important. this part has different goals than the core library which results in measurable app overhead. otoh, If we change code here to save 2 of 4000 allocations, probably more like racing stripes on a yugo.

make sense?

Reply:
ps the annoying dodging or re-preparing statements is something Cassandra 4 driver gets rid of. (you get warning in log now if you accidentally do it). otoh I dont have stomache for another big refactor at the moment because api 4 is very different than 3. good news though is when that occurs we can remove that memorization or have the option to